### ISSUE TITLE
zip output with mixed datatypes

### ISSUE THREAD

**Issue by francoisthire** [nushell/nushell#8785]
### Describe the bug

When the command `zip` is used with mixed datatypes such as a list with a record, a weird output is produced.

### How to reproduce

```nu
 [1,2,3,4] | zip {x:5,y:6,z:7}
```

### Expected behavior

Probably an error message?

### Configuration

| key                | value                                      |
| ------------------ | ------------------------------------------ |
| version            | 0.77.1                                     |
| branch             |                                            |
| commit_hash        |                                            |
| build_os           | linux-x86_64                               |
| build_target       | x86_64-unknown-linux-gnu                   |
| rust_version       | rustc 1.69.0-beta.2 (2602aa81a 2023-03-17) |
| rust_channel       | beta-x86_64-unknown-linux-gnu              |
| cargo_version      | cargo 1.69.0-beta.2 (9880b408a 2023-02-28) |
| build_time         | 2023-03-26 19:39:11 +02:00                 |
| build_rust_channel | release                                    |
| features           | default, zip                               |
| installed_plugins  |                                            |

**Comment by sholderbach** [comment]
Agreed that the resulting nesting levels are more than surprising. (and coming from other languages that the record is not directly iterable here)
```
~> [1,2,3,4] | zip {x:5,y:6,z:7}
╭───┬───────────────────╮
│ 0 │ ╭───┬───────────╮ │
│   │ │ 0 │         1 │ │
│   │ │ 1 │ ╭───┬───╮ │ │
│   │ │   │ │ x │ 5 │ │ │
│   │ │   │ │ y │ 6 │ │ │
│   │ │   │ │ z │ 7 │ │ │
│   │ │   │ ╰───┴───╯ │ │
│   │ ╰───┴───────────╯ │
╰───┴───────────────────╯
```
